Sources/Core/AST/Stmt/ConditionalCompilationStmt.swift

```diff
@@ -94,10 +94,70 @@ public struct ConditionalCompilationStmt: Stmt {
 
   }
 
+  /// A sequence of structured conditions to elaborate a 1 to N operand compilation condition
+  public indirect enum SequenceCondition: Codable {
+    case operand(Condition)
+    case and(Condition, SequenceCondition)
+    case or(Condition, SequenceCondition)
+
+    public enum ConditionKind: String {
+      case holdOnly
+      case skipMain
+      case skipElse
+    }
+
+    /// Visit the SequenceCondition tail to calculate the right case for executing the right branch
+    private func unroll(for factors: ConditionalCompilationFactors, condCase: ConditionKind) -> Bool
+    {
+      switch self {
+      case .and(let cond, let seq):
+        switch condCase {
+        case .holdOnly:
+          return cond.holds(for: factors) && seq.unroll(for: factors, condCase: condCase)
+        case .skipMain:
+          return cond.mayNotNeedParsing && !cond.holds(for: factors)
+            && seq.unroll(for: factors, condCase: condCase)
+        case .skipElse:
+          return cond.mayNotNeedParsing && cond.holds(for: factors)
+            && seq.unroll(for: factors, condCase: condCase)
+        }
+      case .or(let cond, let seq):
```

Comment:
```suggestion

      case .or(let cond, let seq):
```

I would add a newline here for legibility.